doc/source/security/enabling-master-key.rst

```diff
@@ -0,0 +1,206 @@
+.. _enabling-master-key:
+
+==============================================================================
+Enabling Master Key Encryption
+==============================================================================
+
+.. _enabling-vault:
+
+Enabling the Keyring Vault
+---------------------------
+
+To enable Master key vault encryption, the user must have
+`SUPER
+<https://dev.mysql.com/doc/refman/5.7/en/privileges-provided.html#priv_super>`_
+privileges.
+
+Add the following statements to my.cnf:
+
+.. code-block:: MySQL
+
+    [mysqld]
+    early-plugin-load="keyring_valut=keyring_valut.so"
+    loose-keyring_value_config="/home/mysql/keyring_vault.conf"
+
+Restart MySQL.
+
+Encrypting a General Tablespace
+-------------------------------------------
+
+.. _enabling-file-per-table:
+
+The InnoDB system tablespace is encrypted by using master key encryption. The
+server must be started with the `initialize
+<https://dev.mysql.com/doc/refman/5.7/en/server-options.html#option_mysqld_initialize>`_ option.
+
+If the variable :variable:`innodb_sys_tablespace_encrypt` is set to OFF and the
+server has been started with ``initialize=ON``, you may create an encrypted
+table
+as follows:
+
+.. code-block:: guess
+
+    mysql> CREATE TABLE t1 (id INT) ENCRYPTION='Y';
+
+To encrypt a file-per-table tablespace, add the `ENCRYPTION` option.
+
+.. code-block:: MySQL
+
+    mysql> ALTER TABLE t1 ENCRYPTION='Y';
+
+You can also disable encryption for a file-per-table tablespace, set the
+encryption to `N`.
+
+.. code-block:: MySQL
+
+    mysql> ALTER TABLE t1 ENCRYPTION='N';
+
+.. _encrypting-system-tablespace:
+
+Encrypting System Tablespace
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+To enable system tablespace encryption, edit the my.cnf file with the following:
+
+* Add ``innodb_sys_tablespace_encrypt``
+* Edit the ``innodb_sys_tablespace_encrypt`` value to "ON"
+
+System tablespace encryption can only be enabled with the ``--initialize``
+option. You cannot encrypt existing tables in the System tablespace.
+.. _encrypting-undo-log:
+
+Encrypting the Undo Log Data
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+You enable encryption for an undo log data by adding the following line to the
+my.cnf file:
+
+.. code-block:: MySQL
+
+  [mysqld]
+  innodb_undo_log_encrypt=ON
+
+.. _encrypting-binary-logs:
+
+Encrypting Binary Logs
+-----------------------
+
+:Availability: This feature is **Experimental** quality.
+
+Enable encryption of binary logs by running the following statement:
+
+.. code-block:: MySQL
+
+    binlog_encryption=ON
```

Comment:
in 5.7 binlog encryption is enabled by enabling –encrypt-binlog=1. –master_verify_checksum and –binlog_checksum both have to be ON (see https://www.percona.com/blog/2018/03/08/binlog-encryption-percona-server-mysql/ for more details)